R/test-path.R

```diff
@@ -1,23 +1,42 @@
-#' Locate file in testing directory.
+#' Locate file in testing directory
 #'
-#' This function is designed to work both interactively and during tests,
-#' locating files in the `tests/testthat` directory
+#' Many tests require some external file (e.g. a `.csv` if you're testing a
+#' data import function) but the working directory varies depending on the way
+#' that you're running the test (e.g. interactively, with `devtools::tests()`,
+#' or with `R CMD check`). `test_path()` understands these variations and
+#' automatically generates a path relatively to `tests/testthat`, regardless of
+#' where that directory might reside relative to the current working directory.
 #'
-#' @param ... Character vectors giving path component.
+#' @param ... Character vectors giving path components.
 #' @return A character vector giving the path.
 #' @export
+#' @examples
+#' \dontrun{
+#' test_path("foo.csv")
+#' test_path("data", "foo.csv")
+#' }
 test_path <- function(...) {
   if (is_testing() && !isTRUE(getOption("testthat_interactive"))) {
-    if (missing(...)) {
-      "."
-    } else {
-      file.path(...)
-    }
+    base <- NULL
+  } else if (pkgload::is_loading()) {
+    # Probably called from a helper file
+    base <- NULL
   } else {
-    base <- file.path("tests", "testthat")
+    base <- "tests/testthat"
+
     if (!dir.exists(base)) {
-      abort("Can't find `tests/testthat/` in current directory.")
+      cli::cli_abort("Can find {.path {base}}.")
```

Comment:
Another example of https://github.com/r-lib/cli/issues/587 where we show a `.path` fragment that doesn't actually exist, but it will seem clickable.